### PR TITLE
Periodically refresh cloud metadata in k8s reporter

### PR DESCRIPTION
**Pull request by ramanan-ravi**

### CODE REVIEW

**Review by gnmahanth — APPROVED** [review]

File: deepfence_agent/tools/apache/scope/probe/kubernetes/service.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (s *service) GetNode() report.TopologyNode {
 	return report.TopologyNode{
 		Metadata: metadata,
 		Parents: &report.Parent{
-			CloudProvider:     cloudProviderNodeId,
+			//CloudProvider:     cloudProviderNodeId,
```
Comment (noboruma) [inline deepfence_agent/tools/apache/scope/probe/kubernetes/service.go:78]: Why are we removing this one?

File: deepfence_agent/tools/apache/scope/probe/kubernetes/kubernetes_cluster.go [hunk 2]
```diff
@@ -10,35 +11,75 @@ import (
...
+
+func (k *kubernetesCluster) GetTopology() report.Topology {
+	k.RLock()
+	defer k.RUnlock()
+
+	return k.k8sTopology
 }
 
 func (k *kubernetesCluster) GetNode() report.TopologyNode {
+	cloudMetadata := cloud_metadata.GetCloudMetadata()
```
Comment (noboruma) [inline deepfence_agent/tools/apache/scope/probe/kubernetes/kubernetes_cluster.go:66]: We could avoid doing this if we know already we are not "private_cloud"

File: deepfence_agent/tools/apache/scope/probe/kubernetes/service.go (hunk 1 above)
Comment (ramanan-ravi) [reply deepfence_agent/tools/apache/scope/probe/kubernetes/service.go:78]: I don't see it being used in the console.
It is not a global variable anymore.
We can add if required in future.

File: deepfence_agent/tools/apache/scope/probe/kubernetes/kubernetes_cluster.go (hunk 2 above)
Comment (ramanan-ravi) [reply deepfence_agent/tools/apache/scope/probe/kubernetes/kubernetes_cluster.go:66]: Only after this we know whether we are in a known public cloud or not.

**Pull request merged**